File: VB/NorthwindModel.vb
Imports System
Imports System.Collections.Generic
Imports T264297.NwindServiceReference

Namespace T264297

    Friend Class NorthwindModel

        Private context As NorthwindEntities = New NorthwindEntities(New Uri("http://services.odata.org/V3/Northwind/Northwind.svc"))

        Public Function Categories() As List(Of Category)
            Return context.Categories.Expand("Products").ToList()
        End Function
    End Class
End Namespace

File: CS/NorthwindModel.cs
using NorthwindModel;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Net;
using System.Runtime.Remoting.Contexts;
using System.Text;

namespace T264297 {
    public class NorthwindModel {
        public IEnumerable<Category> Categories()
        {
            ServicePointManager.Expect100Continue = true;
            ServicePointManager.SecurityProtocol = SecurityProtocolType.Tls12;
            NorthwindEntities context = new NorthwindEntities(new Uri("https://services.odata.org/V3/Northwind/Northwind.svc"));
            IEnumerable<Category> categories = context.Categories.Execute();
            return categories;
        }
    }
}
